lib/src/drawConfigContext.dart

```diff
@@ -87,31 +126,68 @@ class DrawConfigContext {
     if (primaryFile.exists() == false) {
       primaryFile = new File(this._globalConfigPath.toString());
     }
+  }
 
-    //Parse the ini file.
-    _customConfig = new Config.fromStrings(primaryFile.readAsLinesSync());
-    _initializeAttributes();
+  void _fieldInitializer(type, params) {
+    params.forEach((value) => _initializeField(type, value));
   }
 
   /// Initialize the attributes of the configuration object using the ini file.
-  void _initializeAttributes() {
-    _shortURL = _fetchDefault('short_url');
-    checkForUpdates = _configBool(_fetchDefault('check_for_updates'));
-
-    oauthUrl = _fetch('oauth_url');
-    redditUrl = _fetch('reddit_url');
-
-    //The use of null aware operators here is to give highest precedence to
-    //passed in values to the constructor.
-    clientId ??= _fetchOrNotSet('client_id');
-    clientSecret ??= _fetchOrNotSet('client_secret');
-    httpProxy ??= _fetchOrNotSet('http_proxy');
-    httpsProxy ??= _fetchOrNotSet('https_proxy');
-    _redirectUri ??= _fetchOrNotSet('redirect_uri');
-    refreshToken ??= _fetchOrNotSet('refresh_token');
-    password ??= _fetchOrNotSet('password');
-    userAgent ??= _fetchOrNotSet('user_agent');
-    username ??= _fetchOrNotSet('username');
+  void _initializeField(String type, String param) {
+    if (type == kShortUrl) {
+      _shortURL = _fetchDefault('short_url');
+    } else if (type = kCheckForUpdates) {
+      checkForUpdates = _configBool(_fetchDefault('check_for_updates'));
+    } else if (type = kKind) {
+      //TODO(kc3454): Learn how to do this one.
+    } else if (type = kOptionalField) {
+      String value = _fetchOrNotSet(param);
+      if (value != null) {
+        switch (param) {
+          case 'cliend_id':
+            clientId = value;
+            break;
+          case 'client_secret':
+            clientSecret = value;
+            break;
+          case 'http_proxy':
+            httpProxy = value;
+            break;
+          case 'https_proxy':
+            httpsProxy = value;
+            break;
+          case 'redirect_uri':
+            _redirectUri = value;
+            break;
+          case 'refresh_token':
+            refreshToken = value;
+            break;
+          case 'password':
+            password = value;
+            break;
+          default:
+            throw new DRAWInternalError(
+                'Param $param does not exist in the fieldList for $type');
+            break;
+        }
+      }
+    } else if (type = kRequiredField) {
+      String value = _fetch(param);
+      if (value != null) {
+        switch (param) {
+          case 'oauth_url':
+            oauthUrl = value;
+            break;
+          case 'reddit_url':
+            redditUrl = value;
+            break;
+          default:
+            throw new DRAWInternalError(
+                'Param $param does not exist in the fieldList for $type');
```

Comment:
also, is this not an error on the part of the user? If so, don't throw a DRAWInternalError.